src/htcp.cc

```diff
@@ -1533,7 +1533,8 @@ htcpQuery(StoreEntry * e, HttpRequest * req, CachePeer * p)
     stuff.S.method = sb.c_str();
     stuff.S.uri = (char *) e->url();
     stuff.S.version = vbuf;
-    HttpStateData::httpBuildRequestHeader(req, e, nullptr, &hdr, flags);
+    const auto peer = cbdataReferenceValid(p) ? p : nullptr;
```

Comment:
We should not check `p` validity (here or at all) because we dereference `p` above. I suggest using without checking in this case.

Reply:
Ok.